### PR TITLE
Correction of decoding pvX_mode(_label) and considering number of mppt and number of string of PV-modules in sensor filters

### PR DESCRIPTION
**Pull request by gowima**
1. Decoding pv_modes: ByteH has to exchanged with ByteL and vice versa. 

2. Filters for sensors
The register ppv*, mppt* and the registers ipv*, vpv* have to be filtered in different ways.
Solution based on experiences with 29K9ET:
- ppv* and *mppt are related to number of MPPT's
- ipv's and vpv's are related to the number of strings of PV-modules

3. Added for my convenience method __str__ for Class ET.

### CODE REVIEW

File: goodwe/et.py [hunk 1]
```diff
@@ -40,14 +41,21 @@ class ET(Inverter):
...
-        EnumL("pv2_mode_label", 35120, PV_MODES, "PV2 Mode", Kind.PV),
-        ByteH("pv1_mode", 35120, "PV1 Mode code", "", Kind.PV),  # h
-        EnumH("pv1_mode_label", 35120, PV_MODES, "PV1 Mode", Kind.PV),
+# =============================================================================
+# Changed gowima: Byte order in the implementation of ByteH|L and EnumH|L
+#                 is big endian, which is in accordance to the GoodWe Modbus 
+#                 register documentation. So the byte sequence left to right 
+#                 in a register is H - L. Modes are stored as a byte sequence
+#                 pv4_mode ... pv1_mode. ---> ByteH/ByteL have to be exchanged 
+#                 pairwase for mode 4, 3 and 2, 1.
```
Comment (mletenay) [inline goodwe/et.py:50]: There was conflict with another PR merged today, so I took the liberty and took this fix and committed it directly.
Please rebase/resolve the conflict, I will review the rest tomorrow.
And thanks for noticing and fixing this !

File: goodwe/et.py [hunk 2]
```diff
@@ -459,12 +521,99 @@ async def read_device_info(self):
...
+                           "pv11", "pv12", "pv13", "pv14", "pv15", "pv16",
+                           "mppt4", "mppt5", "mppt6", "mppt7", "mppt8"]
+        # for 4 MPPTs with 2 strings per MPPT
+        mppt4_2string_f = ["pv9", "pv10", "pv11", "pv12", "pv13", "pv14", "pv15", "pv16",
+                           "mppt5", "mppt6", "mppt7", "mppt8"]
+
+        # set filter lists based on parameters of inverter model
+
+        # implicitly 4 mppts come with 2 strings per mppt
+        # TODO: CHECK if device with 4mppt and 1 string per mppt exists
```
Comment (mletenay) [inline goodwe/et.py:583]: It seems it does exist - GE's white label GEH model.
https://www.gesolarinverter.com/images/FILES/DOWNLOADS/1-GLOBAL/1-DATASHEETS/GEH%205-10kW_G.pdf

File: goodwe/model.py [hunk 3]
```diff
@@ -14,11 +14,17 @@
                        "EHU", "EHR", "HSB",  # ET
                        "ESN", "EMN", "ERN", "EBN", "HLB", "HMB", "HBB", "SPN"]  # ES Gen 2
 
-MPPT3_MODELS = ["MSU", "MST", "PSC", "MSC"]
-
-MPPT4_MODELS = ["HSB",
+MPPT3_MODELS = ["MSU", "MST", "PSC", "MSC",
                 "25KET", "29KET", "29K9ET", "30KET"]
```
Comment (mletenay) [inline goodwe/model.py:18]: Would be great if you could provide raw response samples so we can add them to unit tests.
We do not have 29K ET samples yet and these 29KET and 30KET are most likely just wrong, I added them when first found ET >15K models and did not know how goodwe encodes it in serialnr. 29K9ET is the correct one.

**Comment by gowima** [comment]
Here is a listing (txt) of 29K9ET with the script (zip) and necessary data (json) to produce it.

The hardware setup of the inverter is:
- 3 active MPPT, 
- string 1 connected to MMPT 1 (ipv1, vpv1),
- string 2 connected to MPPT 2 (ipv3, vpv3),
- string 3 connected to MPPT 2 (ipv5, vpv5),
- neither battery 1 nor battery 2 are connected

The firmware for DSP and ARM are outdated. We are waiting for the assembly operator to fix that, since we do not (yet) have access to the device configuration.

[goodwe-devel-print.txt](https://github.com/marcelblijleven/goodwe/files/13963426/goodwe-devel-print.txt)
[goodwe-print.json](https://github.com/marcelblijleven/goodwe/files/13963427/goodwe-print.json)
[goodwe-print.zip](https://github.com/marcelblijleven/goodwe/files/13963449/goodwe-print.zip)

**Pull request closed without merging**